nemo/collections/nlp/models/machine_translation/mt_enc_dec_model.py

```diff
@@ -193,6 +194,21 @@ def __init__(self, cfg: MTEncDecModelConfig, trainer: Trainer = None):
             max_delta_length=cfg.max_generation_delta,
         )
 
+        # tie embedding weights
+        if self.shared_embeddings:
+            # validate vocabulary size and embedding dimension
+            if (
+                self.encoder.embedding.token_embedding.weight.shape
+                != self.decoder.embedding.token_embedding.weight.shape
+            ):
+                raise ValueError(
+                    f"Cannot tie encoder and decoder embeddings due to mismatch in embedding sizes "
+                    f"(num_embeddings, embedding_dim): {self.encoder.embedding.token_embedding.weight.shape} (encoder) "
+                    f"{self.decoder.embedding.token_embedding.weight.shape} (decoder)"
+                )
+
+            self.encoder.embedding.token_embedding.weight = self.decoder.embedding.token_embedding.weight
```

Comment:
This seems to only tie token_embeddings, what happens in case `learn_positional_encodings` is set?

Reply:
Good point. Let me check that.

Reply:
Cool